src/celeritas/mat/MaterialParams.cc

```diff
@@ -109,13 +129,29 @@ MaterialParams::from_import(ImportData const& data)
  */
 MaterialParams::MaterialParams(Input const& inp)
 {
+    CELER_EXPECT(!inp.elements.empty());
     CELER_EXPECT(!inp.materials.empty());
 
     ScopedMem record_mem("MaterialParams.construct");
 
-    // Build elements and materials on host.
+    // Build input data on host
     HostValue host_data;
 
+    isotope_data_ = !inp.materials.empty() ? true : false;
+
+    if (isotope_data_)
+    {
+        // Isotopes
+        std::vector<Label> isot_labels(inp.isotopes.size());
+        for (auto i : range(inp.isotopes.size()))
+        {
+            isot_labels[i] = inp.isotopes[i].label;
+            this->append_isotope_def(inp.isotopes[i], &host_data);
+        }
+        isot_labels_ = LabelIdMultiMap<IsotopeId>(std::move(isot_labels));
+    }
```

Comment:
The extra conditional here is unnecessary: if no isotopes are provided, the loop will be empty and the labels will be a null-op.